core/config/project_settings.h

```diff
@@ -243,3 +250,27 @@ Variant _GLOBAL_DEF(const PropertyInfo &p_info, const Variant &p_default, bool p
 #define GLOBAL_DEF_RST_NOVAL_BASIC(m_var, m_value) _GLOBAL_DEF(m_var, m_value, true, true, true)
 
 #define GLOBAL_DEF_INTERNAL(m_var, m_value) _GLOBAL_DEF(m_var, m_value, false, false, false, true)
+
+/////////////////////////////////////////////////////////////////////////////////////////
+// Cached versions of GLOBAL_GET.
+// Cached but uses a typed variable for storage, this can be more efficient.
+// N.B. Variables are prefixed with _ggc_ because compiler is set to report
+// shadowed variable names outside the lambda.
+#define GLOBAL_GET_CACHED(m_type, m_setting_name) ([](const char *p_name) -> m_type {\
```

Comment:
```suggestion
#define GLOBAL_GET_CACHED(m_type, m_setting_name) ([](const char *p_name) -> const m_type& {\
```

I missed this last time; by returning `m_type` we'd be forcing a copy to be made on return. I think this can get optimized out by the compiler (for trivial types anyway), but it still might be more appropriate to return the reference?

Reply:
I think this originally was a `const ref`, then we changed to by value because of the race condition (`local_var` could be changed by another thread after returning it).

Now it's a temporary it could be const ref again, but it might be a compile error because it's a stack variable. I'll try it though. As you say it's probably compiled out.

Reply:
Ohh you're right, it totally needs to be a copy!